src/webgpu/api/validation/encoding/cmds/debug.spec.ts

```diff
@@ -0,0 +1,143 @@
+export const description = `
+API validation test for debug groups and markers
+
+Test Coverage:
+  - For each encoder type (GPUCommandEncoder, GPUComputeEncoder, GPURenderPassEncoder,
+  GPURenderBundleEncoder):
+    - Test that all pushDebugGroup must have a corresponding popDebugGroup
+      - Push and pop counts of 0, 1, and 2 will be used.
+      - An error must be generated for non matching counts.
+    - Test calling pushDebugGroup with an empty string.
+    - Test inserting a debug marker with an empty string.
+    - Test inserting a debug marker with a non-empty string.
+`;
+
+import { poptions, params } from '../../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { assert } from '../../../../../common/framework/util/util.js';
+
+import { ValidationTest } from './../../validation_test.js';
+
+type Encoders = GPUCommandEncoder | GPUProgrammablePassEncoder;
+const kEncoderTypes = ['non-pass', 'compute pass', 'render pass', 'render bundle'] as const;
+type EncoderTypes = typeof kEncoderTypes[number];
+
+class F extends ValidationTest {
+  #commandEncoder: GPUCommandEncoder | undefined = undefined;
```

Comment:
Hm, what is this syntax? I'm unfamiliar with it and having a hard time looking it up.

Reply:
It is ECMAScript's private fields syntax. Unlike typescript's private keyword, the fields aren't accessible outside of typescript. I can switch it to private to match the rest of the code base.

reference:  https://devblogs.microsoft.com/typescript/announcing-typescript-3-8/#ecmascript-private-fields

Reply:
Cool! Fine with me to use the new syntax, but we should look at the generated output and make sure it works on various browsers. (If it outputs the new syntax verbatim, then we need to make sure at least the latest dev versions of Firefox and Safari support it. Otherwise, we should set up Babel to translate to older syntax.)

Reply:
I checked the generated output. It looks like the typescript compiler emulates private fields using a WeakMap. WeakMap has pretty good cross-browser support, although it is probably slower than a private field.

Reply:
Hm, it's too bad that the emulation is a bit complex. I think we should just keep using typescript's private keyword. There isn't much value in true-private in this project since it's not a library, so its types are never exposed outside the project (and the project is all typescript).